File: Python/Class/Inheritance.py
class Animal:
    def __init__(self, name):
        self.name = name

    def sound(self):
        pass


class Dog(Animal):
    def sound(self):
        return "Woof!"


class Cat(Animal):
    def sound(self):
        return "Meow!"


# Creating instances of the classes
animal = Animal("Animal")
dog = Dog("Buddy")
cat = Cat("Whiskers")

# Calling the sound method on the instances
print(animal.sound())  # Output: None (since Animal class has no implementation)
print(dog.sound())     # Output: Woof!
print(cat.sound())     # Output: Meow!

File: C++/inheritance.cpp
#include <iostream>
class Animal {
public:
    void sound() {
        std::cout << "Animal sound" << std::endl;
    }
};
class Dog : public Animal {
public:
    void sound() {
        std::cout << "Woof" << std::endl;
    }
};
int main() {
    Dog dog;
    dog.sound();
    return 0;
}
